dart/lib/src/sentry_client.dart

```diff
@@ -378,12 +378,13 @@ class SentryClient {
   }
 
   SentryEvent _eventWithRemovedBreadcrumbsIfHandled(SentryEvent event) {
-    final exceptions = event.exceptions ?? [];
-    final handled = exceptions.isNotEmpty
-        ? exceptions.first.mechanism?.handled == true
-        : false;
+    final mechanisms =
+        (event.exceptions ?? []).map((e) => e.mechanism).whereType<Mechanism>();
+    final hasNoMechanism = mechanisms.isEmpty;
+    final hasOnlyHandledMechanism =
+        mechanisms.every((e) => (e.handled ?? false));
```

Comment:
The default should still be `true` here otherwise the same bug would happen.

Reply:
So a mechanism that has no `handled` flag (null) is considered as handled?

Reply:
yes, it's only not handled (app crashed) if `handled = false`, everything else is considered that the error was handled (the app didn't crash).